tests/utils/managed_process.py

```
@@ -0,0 +1,239 @@
+            )
+
+        else:
+            with open(self._log_path, "w", encoding="utf-8") as f:
+                self.proc = subprocess.Popen(
+                    self.command,
+                    env=self.env or os.environ.copy(),
+                    cwd=self.working_dir,
+                    stdin=stdin,
+                    stdout=stdout,
+                    stderr=stderr,
+                )
```

Pylint (3.3.7)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)